utilities/ismrmrd_stream_to_hdf5.cpp

```diff
@@ -0,0 +1,148 @@
+                deserializer.deserialize(img);
+                d.appendImage(create_image_series_name(img), img);
+                break;
+            } else if (deserializer.peek_image_data_type() == ISMRMRD::ISMRMRD_CXFLOAT) {
+                ISMRMRD::Image<std::complex<float> > img;
+                deserializer.deserialize(img);
+                d.appendImage(create_image_series_name(img), img);
+                break;
+            } else if (deserializer.peek_image_data_type() == ISMRMRD::ISMRMRD_CXDOUBLE) {
+                ISMRMRD::Image<std::complex<double> > img;
+                deserializer.deserialize(img);
+                d.appendImage(create_image_series_name(img), img);
+                break;
+            } else {
+                throw std::runtime_error("Unknown image type");
+                break;
+            }
+        } else if (deserializer.peek() == ISMRMRD::ISMRMRD_MESSAGE_WAVEFORM) {
+            ISMRMRD::Waveform wfm;
+            deserializer.deserialize(wfm);
+            d.appendWaveform(wfm);
+        } else {
+            std::stringstream ss;
+            ss << "Unknown message type " << deserializer.peek();
+            throw std::runtime_error(ss.str());
+        }
+    }
+}
+
+int main(int argc, char **argv) {
+    // Arguments
+    std::string input_file = "";
+    std::string output_file;
+    std::string groupname;
+    bool use_stdin = false;
+
+    // Parse arguments using boost program options
+    po::options_description desc("Allowed options");
+
+    // clang-format off
+    desc.add_options()
+        ("help,h", "produce help message")
+        ("input,i", po::value<std::string>(&input_file),"Binary input file")
+        ("output,o", po::value<std::string>(&output_file)->required(),"ISMRMRD HDF5 output file")
+        ("use-stdin", po::bool_switch(&use_stdin), "Use stdout for output")
+        ("group,g", po::value<std::string>(&groupname)->default_value("dataset"), "group name");
+    // clang-format on
+
+    po::variables_map vm;
+    try {
+        po::store(po::parse_command_line(argc, argv, desc), vm);
+        po::notify(vm);
+    } catch (boost::wrapexcept<po::required_option> &e) {
+        std::cerr << "Error: " << e.what() << std::endl;
+        std::cerr << desc << std::endl;
+        return 1;
+    }
+
+    if (vm.count("input") && use_stdin) {
+        std::cerr << "Error: Cannot specify both output file and use-stdout" << std::endl;
+        return 1;
+    }
+
+    if (vm.count("help")) {
+        std::cerr << desc << "\n";
+        return 1;
+    }
+
+    if (vm.count("input")) {
+        std::ifstream is(input_file.c_str(), std::ios::binary);
+        if (!is) {
+            std::cerr << "Error: Could not open input file " << input_file << std::endl;
+            return 1;
+        }
+        convert_stream_to_hdf5(output_file, groupname, is);
+    } else if (use_stdin) {
+        ISMRMRD::set_binary_io();
+        convert_stream_to_hdf5(output_file, groupname, std::cin);
```

Comment:
I think we should make sure we have consumed the entire stream at this point.

Reply:
Good suggestion. Guess the question is what is the behavior if there is more data in the stream? It would happen if there we hit close before the end. Should we start over and keep consuming? Not sure that makes sense, so an error?

Reply:
Safest is probably to raise an error.

Reply:
Done